include/sensors/core/tasks/humidity_task.hpp

```diff
@@ -0,0 +1,165 @@
+#pragma once
+
+#include "can/core/can_writer_task.hpp"
+#include "can/core/ids.hpp"
+#include "can/core/messages.hpp"
+#include "common/core/bit_utils.hpp"
+#include "common/core/logging.hpp"
+#include "common/core/message_queue.hpp"
+#include "pipettes/core/messages.hpp"
+#include "sensors/core/humidity_sensor.hpp"
+#include "sensors/core/utils.hpp"
+
+namespace humidity_task {
+
+using namespace humidity_utils;
+using namespace can_ids;
+
+template <message_writer_task::TaskClient CanClient>
+struct HumidityReadingCallback {
+    CanClient &can_client;
+
+    void operator()(const pipette_messages::MaxMessageBuffer &buffer) {
+        uint16_t data = 0x0;
+        const auto *iter = buffer.cbegin();
```

Comment:
should iter be declared a const if it's getting reassigned? what am I missing here?

Reply:
classic cdecl stuff - when defining this `const auto *iter`, it means "an iterator dereferencable to a `const` value whose value is deduced". The way to spell "an iterator whose value cannot be changed" is like `auto *const iter` or something